pilot/pkg/security/model/authentication.go

```diff
@@ -205,8 +209,39 @@ func ConstructValidationContext(rootCAFilePath string, subjectAltNames []string)
 	return ret
 }
 
+// ConstructSANWithTrustDomain constructs MatchSubjectAltNames to be used in DefaultValidationContext by merging
+// Exact SAN matcher user has supplied at Gateway and Prefix SAN matcher for trust domain validation
+func ConstructSANWithTrustDomain(in []string, push *model.PushContext) []*matcher.StringMatcher {
+	//  initially contains only usersupplied SAN
+	userSupplied := util.StringToExactMatch(in)
+	// Early exit if Trust Domain not to be verified
+	if features.SkipValidateTrustDomain.Get() {
+		return userSupplied
+	}
+	trustDomainAliases := push.Mesh.TrustDomainAliases
+	// Add Trust Domain SAN Prefix matching if feature's enabled
+	trustDomain := spiffe.URIPrefix + spiffe.GetTrustDomain()
+
+	tdMatcher := make([]*matcher.StringMatcher, 0, 1+len(trustDomainAliases))
+	// append TD from istiod
+	tdMatcher = append(tdMatcher, &matcher.StringMatcher{
+		MatchPattern: &matcher.StringMatcher_Prefix{Prefix: trustDomain},
+	})
+	// append TD aliases from meshconfig
+	for _, a := range trustDomainAliases {
+		tdMatcher = append(tdMatcher, &matcher.StringMatcher{
+			MatchPattern: &matcher.StringMatcher_Prefix{Prefix: spiffe.URIPrefix + a},
+		})
+	}
+
+	// merge the two
```

Comment:
this comment is not needed and not increasing readability.

Reply:
will remove this